zkevm-circuits/src/evm_circuit/execution/and.rs

```diff
@@ -0,0 +1,328 @@
+use crate::{
+    evm_circuit::{
+        execution::{
+            bus_mapping_tmp::{Block, Call, ExecStep, Transaction},
+            ExecutionGadget,
+        },
+        step::ExecutionResult,
+        table::{FixedTableTag, Lookup},
+        util::{
+            common_gadget::SameContextGadget,
+            constraint_builder::{
+                ConstraintBuilder, StateTransition, Transition::Delta,
+            },
+            Word,
+        },
+    },
+    util::Expr,
+};
+use bus_mapping::{eth_types::ToLittleEndian, evm::OpcodeId};
+use halo2::{arithmetic::FieldExt, circuit::Region, plonk::Error};
+
+#[derive(Clone, Debug)]
+pub(crate) struct AndGadget<F> {
+    same_context: SameContextGadget<F>,
+    a: Word<F>,
+    b: Word<F>,
+    c: Word<F>,
+}
+
+impl<F: FieldExt> ExecutionGadget<F> for AndGadget<F> {
+    const NAME: &'static str = "AND";
+
+    const EXECUTION_RESULT: ExecutionResult = ExecutionResult::AND;
+
+    fn configure(cb: &mut ConstraintBuilder<F>) -> Self {
+        let opcode = cb.query_cell();
+
+        let a = cb.query_word();
+        let b = cb.query_word();
+        let c = cb.query_word();
+
+        cb.stack_pop(a.expr());
+        cb.stack_pop(b.expr());
+        cb.stack_push(c.expr());
+
+        // Because opcode AND, OR, and XOR are continuous, so we can make the
+        // FixedTableTag of them also continuous, and use the opcode delta from
+        // OpcodeId::AND as the delta to FixedTableTag::BitwiseAnd.
+        let tag = FixedTableTag::BitwiseAnd.expr()
+            + (opcode.expr() - OpcodeId::AND.as_u64().expr());
```

Comment:
Use the `opcode` delta from `OpcodeId::AND` as the delta to `FixedTableTag::BitwiseAnd` just as suggested by @Brechtpd [here](https://github.com/appliedzkp/zkevm-circuits/pull/75#discussion_r748296868).